### ISSUE TITLE
SA1407 / IDE0047 conflict

### ISSUE THREAD

**Issue by drieseng** [DotNetAnalyzers/StyleCopAnalyzers#3545]
There's currently no way to avoid both SA1407 or IDE0047 from getting reported for code without precedence issue.

For example:

The following line results in **SA1407** (Arithmetic expressions should declare precedence):
```cs
var x = 10 / 2 % 5;
```

If you add parentheses then **IDE0047** (Parentheses can be removed) is reported:
```cs
var x = (10 / 2) % 5;
```

I reported this as a [bug](https://github.com/dotnet/roslyn/issues/63024) in the .NET Compiler Platform (roslyn) analyzers, but it was rightfully considered as correct behavior.

Is it possible to update the implementation of **SA1407** to no longer report a diagnostic when there's no precedence issue?

**Comment by bjornhellander** [comment]
I checked the code and the handling of `%` seems to be incomplete from day one. We trigger on `%`, but do nothing. Looks a lot like a bug.

**Comment by sharwell** [comment]
Closing this one as **By Design**.
